File: C++/source/Framework/ObjectModel/WaveAsynchronousContextForBootPhases.cpp
/***************************************************************************
 *   Copyright (C) 2005-2010 Vidyasagara Guntaka                           *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/ObjectModel/WaveAsynchronousContextForBootPhases.h"

namespace WaveNs
{

WaveAsynchronousContextForBootPhases::WaveAsynchronousContextForBootPhases (WaveElement *pCaller, WaveAsynchronousCallback pCallback, void *pCallerContext)
    : WaveAsynchronousContext (pCaller, pCallback, pCallerContext),
      m_bootReason             (WAVE_BOOT_FIRST_TIME_BOOT),
      m_rollBackFlag           (false)
{
}

WaveAsynchronousContextForBootPhases::~WaveAsynchronousContextForBootPhases ()
{
}

WaveBootReason WaveAsynchronousContextForBootPhases::getBootReason () const
{
    return (m_bootReason);
}

void WaveAsynchronousContextForBootPhases::setBootReason (const WaveBootReason &bootReason)
{
    m_bootReason = bootReason;
}

bool WaveAsynchronousContextForBootPhases::getRollBackFlag () const
{
    return (m_rollBackFlag);
}

void WaveAsynchronousContextForBootPhases::setRollBackFlag (const bool &rollBackFlag)
{
    m_rollBackFlag = rollBackFlag;
}

}

File: Java/source/com/CxWave/Wave/Framework/ObjectModel/Boot/WaveAsynchronousContextForBootPhases.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.ObjectModel.Boot;

import com.CxWave.Wave.Framework.ObjectModel.WaveElement;
import com.CxWave.Wave.Framework.Utils.Context.WaveAsynchronousContext;
import com.CxWave.Wave.Resources.ResourceEnums.WaveBootReason;

public class WaveAsynchronousContextForBootPhases extends WaveAsynchronousContext
{

    private WaveBootReason m_bootReason;

    public WaveAsynchronousContextForBootPhases (final WaveElement caller, final String callbackMethodName, final Object callerContext)
    {
        super (caller, callbackMethodName, callerContext);
    }

    public WaveBootReason getBootReason ()
    {
        return (m_bootReason);
    }

    public void setBootReason (final WaveBootReason bootReason)
    {
        m_bootReason = bootReason;
    }

}
